controllers/cluster_controller_test.go

```diff
@@ -42,7 +42,7 @@ func TestClusterController(t *testing.T) {
 	}{
```

Comment:
I don't see a test case for management clusters (which shouldn't ever be deleted)

Reply:
Added for vintage and capi clusters.